test_runner/regress/test_lsn_mapping.py

```diff
@@ -108,6 +109,43 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
 
+#
+# Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+#
+# Added as an effort to improve error handling and avoid full anyhow backtrace
+# https://github.com/neondatabase/neon/pull/7949
+#
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+    endpoint = env.endpoints.create_start("main")
```

Comment:
creation of endpoints without a context manager presents an interesting test flakyness point: what if the endpoint is requesting pageserver for something which fails during shutdown.

in general, I'd prefer to see:

```suggestion
    with endpoints.create_start("main") as endpoint:
        # one more identation level
```

because then it describes the scope in which the endpoint exists and can contact safekeepers and pageserver.

however this test does not require anything to exist in the database, we only require that initial_tenant and initial_timeline are present, and we can query the database for timestamp=0 for example, it doesn't matter.